extensions/ql-vscode/src/data-extensions-editor/auto-model-codeml-queries.ts

```diff
@@ -228,3 +243,59 @@ export async function runAutoModelQueries({
     candidates,
   };
 }
+
+/**
+ * generateCandidateFilterPack will create a temporary extension pack.
+ * This pack will contain a filter that will restrict the automodel queries
+ * to the specified candidate methods only.
+ * This is done using the `extensible` predicate "automodelCandidateFilter".
+ * @param language
+ * @param candidateMethods
+ * @returns
+ */
+export async function generateCandidateFilterPack(
+  language: string,
+  candidateMethods: MethodSignature[],
+): Promise<string> {
+  // Pack resides in a temporary directory, to not pollute the workspace.
+  const packDir = (await dir({ unsafeCleanup: true })).path;
+
+  const syntheticConfigPack = {
+    name: "codeql/automodel-filter",
+    version: "0.0.0",
+    library: true,
+    extensionTargets: {
+      [`codeql/${language}-all`]: "*",
```

Comment:
I think you need this, because the extensible predicate `automodelCandidateFilter` is in the query pack, not the library pack. As we introduce "deep validation" of data extensions I think this will start to fail.

In future if you move the automodel queries to their own pack, you'll need to update this pack reference too.
```suggestion
      [`codeql/${language}-queries`]: "*",
```